node/rpc/src/chain_api.rs

```diff
@@ -355,6 +355,37 @@ where
     Ok(ts)
 }
 
+pub(crate) async fn chain_validate_tipset_checkpoints<DB, B>(
+    data: Data<RPCState<DB, B>>,
+    Params(params): Params<ChainValidateTipSetCheckpointsParams>,
+) -> Result<ChainValidateTipSetCheckpointsResult, JsonRpcError>
+where
+    DB: Blockstore + Store + Clone + Send + Sync + 'static,
+    B: Beacon,
+{
+    let () = params;
+
+    let tipset_keys = data
+        .state_manager
+        .chain_store()
+        .heaviest_tipset()
+        .await
+        .unwrap()
+        .key()
+        .cids()
+        .to_vec();
+    let ts = data
+        .state_manager
+        .chain_store()
+        .tipset_from_keys(&TipsetKeys::new(tipset_keys))
```

Comment:
It seems unnecessary to convert the `TipsetKeys` to a vector of cids and then convert it back to a `TipsetKeys`. Why not use the `TipsetKeys` directly?

Reply:
Yes, fixed.

Reply:
Did you push the fix?

Reply:
not yet, will push after addressing remaining comments in this PR